File: app/src/main/java/com/example/getfood/ui/splash/SplashMvpPresenter.java
package com.example.getfood.ui.splash;

import com.example.getfood.ui.base.BaseMvpPresenter;

public interface SplashMvpPresenter<V extends SplashMvpView> extends BaseMvpPresenter<V> {
    void determineIfUserLoggedIn();

    String getVersionName();

    void determineIfUpdateNeeded(String versionName);
}

File: app/src/main/java/com/example/getfood/ui/splash/SplashMvpPresenter.kt
package com.example.getfood.ui.splash

import com.example.getfood.ui.base.BaseMvpPresenter

interface SplashMvpPresenter<V : SplashMvpView?> : BaseMvpPresenter<V> {
    fun determineIfUserLoggedIn()
    val versionName: String
    fun determineIfUpdateNeeded(versionName: String?)
}